SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -280,4 +281,178 @@ bool SpatialCommandUtils::HasDevLoginTag(const FString& DeploymentName, bool bIs
+	}
+}
+
+bool SpatialCommandUtils::GetProcessName(const FString& PID, FString& OutProcessName)
+{
+	bool bSuccess = false;
+	OutProcessName = TEXT("");
+	const FString TaskListCmd = TEXT("tasklist");
+
+	// Get the task list line for the process with PID 
+	const FString TaskListArgs = FString::Printf(TEXT(" /fi \"PID eq %s\" /nh /fo:csv"), *PID);
+	FString TaskListResult;
+	int32 ExitCode;
+	FString StdErr;
+	bSuccess = FPlatformProcess::ExecProcess(*TaskListCmd, *TaskListArgs, &ExitCode, &TaskListResult, &StdErr);
+	if (ExitCode == 0 && bSuccess)
+	{
+		FRegexPattern ProcessNamePattern(TEXT("\"(.+?)\""));
+		FRegexMatcher ProcessNameMatcher(ProcessNamePattern, TaskListResult);
+		if (ProcessNameMatcher.FindNext())
+		{
+			OutProcessName = ProcessNameMatcher.GetCaptureGroup(1 /* Get the Name of the process, which is the first group. */);
+
+			return true;
+		}
+	}
+
+	UE_LOG(LogSpatialCommandUtils, Warning, TEXT("Failed to get the name of the process that is blocking the required port."));
+
+	return false;
+}
+
+bool SpatialCommandUtils::TryKillProcessWithPID(const FString& PID)
+{
+	int32 ExitCode;
+	FString StdErr;
+
+#if PLATFORM_WINDOWS
+	const FString KillCmd = TEXT("taskkill");
+	const FString KillArgs = FString::Printf(TEXT("/F /PID %s"), *PID);
+#elif PLATFORM_MAC
+	const FString KillCmd = FPaths::Combine(SpatialGDKServicesConstants::KillCmdFilePath, TEXT("kill"));
+	const FString KillArgs = FString::Printf(TEXT("%s"), *PID);
+#endif
+
+	FString KillResult;
+	bool bSuccess = FPlatformProcess::ExecProcess(*KillCmd, *KillArgs, &ExitCode, &KillResult, &StdErr);
+	bSuccess = bSuccess && ExitCode == 0;
+	if (!bSuccess)
+	{
+		UE_LOG(LogSpatialCommandUtils, Error, TEXT("Failed to kill process with PID %s. Error: %s"), *PID, *StdErr);
+	}
+
+	return bSuccess;
+}
+
+bool SpatialCommandUtils::GetProcessInfoFromPort(int32 Port, FString& OutPid, FString& OutState, FString& OutProcessName)
+{
+#if PLATFORM_WINDOWS
+	const FString Command = FString::Printf(TEXT("netstat"));
+	// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+	const FString Args = TEXT("-n -o -a");
+#elif PLATFORM_MAC
+	const FString Command = FPaths::Combine(SpatialGDKServicesConstants::LsofCmdFilePath, TEXT("lsof"));
+	// -i:Port list the processes that are running on Port
+	const FString Args = FString::Printf(TEXT("-i:%i"), Port);
+#endif
+
+	FString Result;
+	int32 ExitCode;
+	FString StdErr;
+
+	bool bSuccess = FPlatformProcess::ExecProcess(*Command, *Args, &ExitCode, &Result, &StdErr);
+
+	if (ExitCode == 0 && bSuccess)
+	{
+#if PLATFORM_WINDOWS
+		// Get the line of the netstat output that contains the port we're looking for.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(.*?:%i.)(.*)( [0-9]+)"), Port));
+#elif PLATFORM_MAC
+		// Get the line that contains the name, pid and state of the process.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(\\S+)( *\\d+).*(\\(\\S+\\))")));
```

Comment:
as an idea: could we define the regex inside the SpatialGDKServicesConstants? that way we can get rid of the the  `#if` here and still separate it based on platform. just need to add a comment to explain what it's for / what's it's looking for

Edit: ah we need to pass in the port for windows... could maybe inline it as a function? or maybe it's overkill for this. I don't feel strongly about this